nexus/db-queries/src/db/datastore/image.rs

```diff
@@ -78,7 +178,48 @@ impl DataStore {
             AsyncInsertError::DatabaseError(e) => {
                 public_error_from_diesel_pool(
                     e,
-                    ErrorHandler::Conflict(ResourceType::Image, name.as_str()),
+                    ErrorHandler::Conflict(
+                        ResourceType::ProjectImage,
+                        name.as_str(),
+                    ),
+                )
+            }
+        })?;
+        Ok(image)
+    }
+
+    pub async fn project_image_promote(
+        &self,
+        opctx: &OpContext,
+        authz_silo: &authz::Silo,
+        project_image: ProjectImage,
+    ) -> UpdateResult<Image> {
+        let silo_id = authz_silo.id();
+        let image: Image = project_image.into();
+        let name = image.name().clone();
+
+        opctx.authorize(authz::Action::CreateChild, authz_silo).await?;
```

Comment:
The process of promoting an image is essentially the same as creating a silo image.

Reply:
I'm realizing this is also missing permissions for the project image itself.